src/main/java/net/minecraftforge/items/IItemHandlerContainer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+
+/**
+ * Declares an item handler as being aware of container slot interactions.
+ */
+public interface IItemHandlerContainer
+{
+    /**
+     * Queries if the given slot is acceptable for the inventory
+     * @param slot The slot number within the inventory
+     * @param stack The stack being tested
+     * @return True if the provided stack is acceptable
+     */
+    boolean isItemValidForSlot(int slot, ItemStack stack);
+
+    /**
+     * Gets the stack size limit allowed by the given slot
+     * @param slot The slot number within the inventory
+     * @return The stack size limit
+     */
+    int getInventoryStackLimit(int slot);
```

Comment:
Done. I did need the @Nullable, since one of my own uses doesn't have a stack to pass into it.